configuration/src/main/java/io/airlift/configuration/ConfigurationFactory.java

```diff
@@ -366,12 +366,14 @@ public void addConfigurationBindingListener(ConfigurationBindingListener listene
         Problems problems = new Problems(monitor);
 
         for (AttributeMetadata attribute : configurationMetadata.getAttributes().values()) {
+            Problems attributeProblems = new Problems(monitor);
```

Comment:
I don't understand how this is different than adding them directly to `problems` (which is what it does at the end)